### ISSUE TITLE
Problem decoding Greyhound Data upon Read

### ISSUE THREAD

**Issue by JSFrias** [hobu/greyhound#44]
I'm trying to work with the data returned by the greyhound read query.  It looks like I'm losing precision somewhere along the line.  Having written my own code I had assumed my struggles were a bug of mine. However I just discovered your own sample python code ( https://github.com/hobu/greyhound/pull/35/files ) that appears to have the same issue. Playing with the settings in this file, I run it to produce this query:

http://[myipaddress]:8080/resource/[MyDemoResource]/read?bounds=[-13634057.2874,4552379.08325,-13634052.2874,4552384.08325]&compress=false

It returns 30813 points from a 5 meter cube of my data. The las file that's generated looks like this:

![image](https://user-images.githubusercontent.com/17951316/28143785-500b950c-671c-11e7-8835-406704cc4353.png)

You can probably see 30-4 points within the bounds of this data in this image above. When I analyze the data itself, there are really only 6 unique values in each of the X, and Y coordinates.  In other words, each point visible in this image is actually about 880 superimposed vertices.

The part that I don't understand is that when I aim Potree at this same area, I see the full point density and it looks nothing like this. It's filled in, and almost solid in appearance until you really zoom in on it.

For what it's worth, this is the point breakdown per depth level (I only have data in 11-21):

11:1,12:2,13:15, 14:61, 15: 255, 16:966, 17:3534, 18:10399, 19:13031, 20:2514, 21:35

My schema:
{"baseDepth":7,"bounds":[-13635560,4545640,-6040,-13623480,4557720,6040],"boundsConforming":[-13635553.321355114,4546474.442458361,-346.13140000000004,-13623524.425497472,4556869.877397481,309.4769],"numPoints":1185748768,"offset":[-13629520,4551680,0],"reprojection":{"in":"","out":"EPSG:3857"},"scale":0.001,"schema":[{"name":"X","size":4,"type":"signed"},{"name":"Y","size":4,"type":"signed"},{"name":"Z","size":4,"type":"signed"},{"name":"Intensity","size":2,"type":"unsigned"},{"name":"ReturnNumber","size":1,"type":"unsigned"},

I feel like I'm missing something fundamental in trying to interpret the returned data. Do you have any ideas what I might be missing?

Thank you.

**Comment by connormanning** [comment]
Can you reproduce this against something public so I can tweak the query around?  I have some ideas but no concrete suggestions without being able to fully reproduce.  For example with http://data.greyhound.io/resource/autzen.

Another thing that you might not be aware of you might like - instead of `/read`, you can use `/static`, which will statically render your query.  Query parameters will be forwarded.  For example: http://data.greyhound.io/resource/autzen/static?bounds=[634950,848860,637300,851210]&depthEnd=10.  This is nice for quick query testing.

**Comment by JSFrias** [comment]
Even though you didn't provide a solution, your answer was enough to inspire me to dig a little deeper.  So I've fixed my issue.  My sample data is mobile LiDAR that I indexed with a scale of .001.  I'm not sure if Greyhound defaults internally to serving data at .01 scale, but I went ahead and provided scale and offsets into my query. I also adjusted my x and y bounds parameters (as your documentation says I should do) like so:

x = (x - offset[0]) / scale[0]
y = (y - offset[1]) / scale[1]

(I'm not providing z in my bounds)

That fixed it.

What's curious is that your documentation specifically says "If scale and offset values are passed, and they are exactly equivalent to those present in the info query, then this results in a no-op transformation on the server since the data is already in the desired local coordinate space under-the-hood"

Yet, that's exactly what I did and it fixed my issue.

Thanks again.
